pkg/handlers/app.go

```diff
@@ -421,50 +425,97 @@ func (h *Handler) RemoveApp(w http.ResponseWriter, r *http.Request) {
+	request := CanInstallAppVersionRequest{}
 	if err := json.NewDecoder(r.Body).Decode(&request); err != nil {
 		response.Error = "failed to parse request body"
 		logger.Error(errors.Wrap(err, response.Error))
 		JSON(w, http.StatusBadRequest, response)
 		return
 	}
 
-	if request.AppSpec == "" {
-		response.IsCompatible = true
-		JSON(w, http.StatusOK, response)
-		return
-	}
+	if request.AppSpec != "" {
+		response.CanInstall = false
 
-	kotsApp, err := kotsutil.LoadKotsAppFromContents([]byte(request.AppSpec))
-	if err != nil {
-		response.Error = "failed to load kots app from contents"
-		logger.Error(errors.Wrap(err, response.Error))
-		JSON(w, http.StatusInternalServerError, response)
-		return
-	}
+		kotsApp, err := kotsutil.LoadKotsAppFromContents([]byte(request.AppSpec))
+		if err != nil {
+			response.Error = "failed to load kots app from contents"
+			logger.Error(errors.Wrap(err, response.Error))
+			JSON(w, http.StatusInternalServerError, response)
+			return
+		}
 
-	if kotsApp != nil {
-		response.IsCompatible = kotsutil.IsKotsVersionCompatibleWithApp(*kotsApp, request.IsInstall)
-	} else {
-		response.IsCompatible = true
+		if kotsApp != nil {
+			response.CanInstall = kotsutil.IsKotsVersionCompatibleWithApp(*kotsApp, request.IsInstall)
+		}
+
+		if !response.CanInstall {
+			response.Error = kotsutil.GetIncompatbileKotsVersionMessage(*kotsApp, request.IsInstall)
+			JSON(w, http.StatusOK, response)
+			return
+		}
 	}
 
-	if !response.IsCompatible {
-		response.Error = kotsutil.GetIncompatbileKotsVersionMessage(*kotsApp, request.IsInstall)
+	if request.AirgapSpec != "" {
+		response.CanInstall = false
+
+		a, err := store.GetStore().GetAppFromSlug(appSlug)
+		if err != nil {
+			response.Error = "failed to get kots app"
+			logger.Error(errors.Wrap(err, response.Error))
+			JSON(w, http.StatusInternalServerError, response)
+			return
+		}
+
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		decoded, gvk, err := decode([]byte(request.AirgapSpec), nil, nil)
+		if err != nil {
+			response.Error = "failed to decode airgap spec"
+			logger.Error(errors.Wrap(err, response.Error))
+			JSON(w, http.StatusInternalServerError, response)
+			return
+		}
+
+		if gvk.Group != "kots.io" || gvk.Version != "v1beta1" || gvk.Kind != "Airgap" {
+			response.Error = fmt.Sprintf("invalid airgap spec gvk: %s", gvk.String())
+			logger.Error(errors.Wrap(err, response.Error))
+			JSON(w, http.StatusInternalServerError, response)
+			return
+		}
+
+		missingPrereqs, err := airgap.GetMissingRequiredVersions(a, decoded.(*kotsv1beta1.Airgap))
+		if err != nil {
+			response.Error = "failed to get release prerequisites"
+			logger.Error(errors.Wrap(err, response.Error))
+			JSON(w, http.StatusInternalServerError, response)
+			return
+		}
+
+		if len(missingPrereqs) > 0 {
+			response.Error = fmt.Sprintf("This airgap bundle cannot be deployed because versions %s are required and must be uploaded first.", strings.Join(missingPrereqs, ", "))
```

Comment:
Same here.
```suggestion
			response.Error = fmt.Sprintf("This airgap bundle cannot be uploaded because versions %s are required and must be uploaded first.", strings.Join(missingPrereqs, ", "))
```